core/domain/classifier_services_test.py

```diff
@@ -0,0 +1,166 @@
+        self.string_classifier.load_examples(self._EXAMPLES)
+
+    def _validate_instance(self, string_classifier):
+        self.assertEquals('_alpha' in dir(self.string_classifier), True)
+        self.assertEquals('_beta' in dir(self.string_classifier), True)
+
+        for d in xrange(self.string_classifier._doc_count):
+            self.assertEquals(
+                len(self.string_classifier._l_dc[d]),
+                len(self.string_classifier._w_dc[d]))
+
+        self.assertEquals(
+            len(self.string_classifier._label_to_id),
+            self.string_classifier._label_count)
+        self.assertEquals(
+            len(self.string_classifier._word_to_id),
+            self.string_classifier._word_count)
+        self.assertEquals(
+            len(self.string_classifier._w_dc),
+            self.string_classifier._doc_count)
+        self.assertEquals(
+            len(self.string_classifier._b_dl),
+            self.string_classifier._doc_count)
+        self.assertEquals(
+            len(self.string_classifier._b_dl[0]),
+            self.string_classifier._label_count)
+        self.assertEquals(
+            len(self.string_classifier._l_dc),
+            self.string_classifier._doc_count)
+        self.assertEquals(
+            len(self.string_classifier._c_dl),
+            self.string_classifier._doc_count)
+        self.assertEquals(
+            len(self.string_classifier._c_dl[0]),
+            self.string_classifier._label_count)
+        self.assertEquals(
+            len(self.string_classifier._c_lw),
+            self.string_classifier._label_count)
+        self.assertEquals(
+            len(self.string_classifier._c_lw[0]),
+            self.string_classifier._word_count)
+        self.assertEquals(
+            len(self.string_classifier._c_l),
+            self.string_classifier._label_count)
+
+    def test_valid_state(self):
+        self.assertEquals(self.string_classifier._label_count, 3)
+        self.assertEquals(self.string_classifier._doc_count, 3)
+        self.assertEquals(self.string_classifier._word_count, 10)
+        self._validate_instance(self.string_classifier)
+
+    def test_add_examples(self):
+        self.string_classifier.add_examples(self._NEW_EXAMPLES)
+        self.assertEquals(self.string_classifier._label_count, 3)
+        self.assertEquals(self.string_classifier._doc_count, 6)
+        self.assertEquals(self.string_classifier._word_count, 37)
+        self._validate_instance(self.string_classifier)
+
+    def test_empty_load(self):
+        self.string_classifier.load_examples([])
+
+    def test_empty_add(self):
+        self.string_classifier.add_examples([])
+
+    def test_model_save_load(self):
+        self.assertEquals(
+            self.string_classifier._doc_count,
+            len(self._EXAMPLES))
+        model = self.string_classifier.to_dict()
+        model['_doc_count'] = 9
+        self.assertEquals(model['_doc_count'], 9)
```

Comment:
This test sanity checks load and save, but also makes sure that deepcopy works. Just an extra line to prove a point that yes, the dictionary has changed, while the next line proves that the model has not changed.